src/Libraries/Microsoft.Extensions.Resilience/Resilience/Internal/ResilienceTagNames.cs

```diff
@@ -5,13 +5,13 @@ namespace Microsoft.Extensions.Resilience.Internal;
 
 internal static class ResilienceTagNames
 {
-    public const string FailureSource = "failure-source";
+    public const string FailureSource = "resilience.failure.source";
 
-    public const string FailureReason = "failure-reason";
+    public const string FailureReason = "resilience.failure.reason";
```

Comment:
based on what I see here https://github.com/dotnet/extensions/blob/main/src/Libraries/Microsoft.Extensions.Resilience/Resilience/Internal/ResilienceMetricsEnricher.cs, we can use `error.type` attribute here (and full exception name)

Reply:
One note ere. These are just additional tags on top of the ones built-in into Polly:
https://www.pollydocs.org/advanced/telemetry.html#metrics

One of the tags is `exception.type` one as defined by:
https://opentelemetry.io/docs/specs/otel/trace/exceptions/


There is an intersection here between `exception.type` and `error.type` and I am wondering how should we handle this.

Just for context:

This is how `HttpResponseMessage` is translated into `FailureResultContext`:
https://github.com/dotnet/extensions/blob/124736421d4e44ce1c5a8d9120edef4621a9c87b/src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/ResilienceHttpClientBuilderExtensions.Resilience.cs#L156

And this is how exceptions are translated:
https://github.com/dotnet/extensions/blob/124736421d4e44ce1c5a8d9120edef4621a9c87b/src/Libraries/Microsoft.Extensions.Resilience/Resilience/Internal/ResilienceMetricsEnricher.cs#L37



Can we trim the `FailureResultContext`

Reply:
thank you for the context!

To the original comment: if polly already reports `exception.type`, is there any reason to report `resilience.failure.reason` at all since it's always an exception type?

> Can we trim the FailureResultContext or maybe just expose add the following tags?

error.summary: will be status code such as 500 for HttpResponseMessage or exception summary for exceptions.
exception.source: applicable only for exceptions (or even don't add anything here)

I'd avoid adding new tags to OTel-defined namespaces - this is against OTel recommendations https://github.com/open-telemetry/opentelemetry-specification/blob/563958cb2bd8529990f19fdce7a5f3643bf63091/specification/common/attribute-naming.md?plain=1#L128

But I believe that if the exception summary has low cardinality, you can put it into `error.type` attribute which is intended to capture all kinds of error codes and fallback to (full) exception type.

We can also introduce `dotnet.exception.source`  attribute and (hopefully) reuse in other parts of .NET and 3rd party libraries.

Reply:
I agree with having just:

- `error.type` - to represent both exception summary and the failure reason for results
- `dotnet.exception.source`

And at initial release we won't add tags for the following properties?

- `FailureResultContext.Source`
- `FailureResultContext.AdditionalInformation`

https://github.com/dotnet/extensions/blob/261b3546ab8280dbecd77efcf063771649397fc3/src/Libraries/Microsoft.Extensions.Resilience/Resilience/FailureResultContext.cs#L13

Maybe we can slim the `FailureResultContext` down then?

Reply:
PR with the changes is up:

#4600